### PR TITLE
Talk about documentation

### PR DESCRIPTION
**Pull request by jaylett**
* what there is
 * how to build it (briefly)
 * some brief text about writing documentation to support the code
   you work on
 * how to build the PDF version of the documentation

### CODE REVIEW

File: documentation/index.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Available documentation
+-----------------------
+
+There are a number of types of documentation available for Xapian.
+
+1. User documentation that explains how people will use Xapian,
+   typically either as a "full worked example", or as a deep dive into
+   a particular feature.
+
+   Most of this documentation lives in the*Xapian user guide*, which
```
Comment (ojwb) [inline documentation/index.rst]: Looks like a missing space ("the*).

File: documentation/index.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+   There is also some developer documentation available elsewhere,
+   particularly in the main Xapian source repository -- look in
+   particular in ``xapian-core/docs``. Some of this is being migrated
+   to the developer guide. (Some is also being migrated to the user
+   guide.)
+
+4. `The Xapian wiki`_ also contains information, including links
+   to articles and talks that people have written about using Xapian
+   in various ways. Although this isn't always official Xapian
+   documentation, it may be helpfuil. Some of it may now be quite out
```
Comment (ojwb) [inline documentation/index.rst]: Typo: "helpfuil"

File: documentation/index.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+   pip install -r requirements.txt
+   make html
+
+For the user guide, you can control which language the examples use by
+setting the ``LANGUAGE`` variable when you call make:
+
+.. code-block:: bash
+
+   make html LANGUAGE=c++
```
Comment (ojwb) [inline documentation/index.rst]: I wonder if we want to start duplicating documentation about these sort of details?  If we just said something like "see `make help` for details of the options available" that would avoid having an extra place to update if we change the details of how the language is selected here.

Comment (jaylett) [reply documentation/index.rst]: I've removed the duplication; in the process I noticed that neither user nor dev guides had amazing documentation. I've fixed the dev guide in another commit in this branch, and I'll open a separate PR for the user guide that makes the two more consistent.

**Comment by ojwb** [comment]
Looks good.  Rebased and autosquashed and pushed.  I tweaked `requirements.txt` to say `Sphinx~=1.8.0` instead of `==` to align it with the user guide.

**Pull request closed without merging**